build/Dockerfile

```diff
@@ -0,0 +1,35 @@
+# --- build smart gateway and agent plugins ---
+FROM registry.access.redhat.com/ubi8 AS builder
+ENV GOPATH=/go
+ENV CORE=/go/src/github.com/infrawatch/sg-core
+ENV WD=/go/src/github.com/infrawatch/sg-agent
+
+WORKDIR $WD
+COPY . $WD/
+
+# dependencies for qpid-proton-c
+COPY build/repos/opstools.repo /etc/yum.repos.d/opstools.repo
+RUN dnf install golang git qpid-proton-c-devel -y --setopt=tsflags=nodocs
+
+# build sg-core
+RUN git clone https://github.com/infrawatch/sg-core $CORE && pushd $CORE && PRODUCTION_BUILD=true ./build.sh && popd
```

Comment:
Maybe this won't really be much of an issue since our build process shouldn't change often, but I was going build a makefile for sg-core and we will have to remember to update this too.

Reply:
Good point. Switching from bash script to makefile is a one line change, so it should not be much of an issue. And for the backward compatibility the build.sh could just call make :).